tests/e2e/taqueria-plugin-taquito.spec.ts

```diff
@@ -0,0 +1,345 @@
+
+                // 3. Verify that contract has been originated on the network
+                expect(stdoutDeploy).toContain("hello-tacos.tz");
+                expect(stdoutDeploy).toContain("hangzhounet");
+                const [,,contractHash] = stdoutDeploy.split("│");
+                smartContractHash = contractHash.trim();
+                expect(contractHash.trim()).toMatch(contractRegex);
+            });
+
+            // 4. Verify that contract has been originated to the network
+            await waitForExpect(async () => {
+                const contract = await tezos.contract.at(smartContractHash);
+                expect(contract.address).toBe(smartContractHash);
+            });
+
+        } catch(error) {
+            throw new Error (`error: ${error}`);
+        }
+
+    });
+
+    test('Verify that taqueria taquito plugin can deploy one contract using deploy {contractName} command', async () => {
+        try {
+            environment = "test";
+            let smartContractHash = "";
+
+            // 1. Copy config.json and michelson contract from data folder to artifacts folder under taqueria project
+            execSync(`cp e2e/data/config-taquito-test-environment.json ${taqueriaProjectPath}/.taq/config.json`);
+            execSync(`cp e2e/data/hello-tacos.tz ${taqueriaProjectPath}/artifacts/`);
+
+            // Sometimes running two deploy commands in a short period of time cause an issue
+            // More details about the error - "counter %x already used for contract %y"
+            // It is applicable only for auto-tests that run very fast
+            // Uses retry mechanism to avoid test to fail
+
+            await waitForExpect(() => {
+
+                // 2. Run taq deploy ${contractName} on a selected test network described in "test" environment
+                const [,,,stdoutDeploy] = execSync(`taq deploy hello-tacos.tz -e ${environment}`, {cwd: `./${taqueriaProjectPath}`}).toString().trim().split(/\r?\n/);
+
+                // 3. Verify that contract has been originated on the network
+                expect(stdoutDeploy).toContain("hello-tacos.tz");
+                expect(stdoutDeploy).toContain("hangzhounet");
+                const [,,contractHash] = stdoutDeploy.split("│");
+                smartContractHash = contractHash.trim();
+                expect(contractHash.trim()).toMatch(contractRegex);
+            });
+
+            // 4. Verify that contract has been originated to the network
+            await waitForExpect(async () => {
+                const contract = await tezos.contract.at(smartContractHash);
+                expect(contract.address).toBe(smartContractHash);
+            });
+
+        } catch(error) {
+            throw new Error (`error: ${error}`);
+        }
+
+    });
+
+    test('Verify that taqueria taquito plugin can deploy one contract using originate {contractName} command', async () => {
```

Comment:
Since this test is only testing that the originate alias is mapped to the deploy command I think you can get the same verification that originate will work by not deploying a contract but comparing the output of the `taq deploy --help` and `taq originate --help` and making sure they are the same. The previous test verifies that the contract can be deployed/originated and this one shouldn't have to test that again

Reply:
Let's ask @mweichert I am just afraid that testing help is not a functional test to cover it. I will post the question into discussion.

Reply:
I wouldn't have caught this. Good thoughts Alex.

As the tests probably take a long time to run, it probably does make sense to optimize things to take the least amount of time as possible.

Nonetheless, given that this works, I'd probably suggest we let this go for now, create an issue to come back to this, and add a todo comment with a link to the issue.

In the issue, we can list a few options we're considering:
- Comparing the output of `taq deploy --help` and `taq originate --help`
- Grepping `taq deploy --help`
- Parsing the `state.json` file
- Parsing the pluginInfo response

Reply:
Added issue to make this test - https://github.com/ecadlabs/taqueria/issues/379